### PR TITLE
Add a recipe for globus-cli

### PR DESCRIPTION
**Pull request by xylar**
Add a conda-forge recipe for [Globus CLI](https://globus.github.io/globus-cli)

### CODE REVIEW

File: recipes/globus-cli/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://github.com/globus/globus-cli/archive/{{ version }}.tar.gz
+  sha256: e577b1e53e5f2265fa1f08549e26c8bdf6dbc90c22722dd46277db08c13ac38e
+
+build:
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
```
Comment (chrisburr) [inline recipes/globus-cli/meta.yaml]: [You need to specify `entry_points` here.](https://conda.io/projects/conda-build/en/latest/source/resources/define-metadata.html#python-entry-points)

**Comment by chrisburr** [comment]
It take a little while for new packages to appear in the CDN. I think it refreshes every 20 minutes but I might be misremembering the number.

File: recipes/globus-cli/meta.yaml (hunk 1 above)
Comment (xylar) [reply recipes/globus-cli/meta.yaml]: Okay, thanks.  I'll look into it.

Comment (xylar) [reply recipes/globus-cli/meta.yaml]: Thanks again for your help.  It should be taken care of now.

**Comment by xylar** [comment]
@lukaszlacinski, are you okay with being a maintainer of this recipe, too?

Also, do you know if `globus-cli` should work under Windows?  If so, could you take a look at the appveyor errors and see if they look familiar to you?

**Comment by xylar** [comment]
Hmm, taking a closer look, it's not obvious to me whether the problem with the Windows build has anything to do with the `globus-cli` source. It could be the recipe.  @chrisburr, any thoughts?

**Comment by xylar** [comment]
I strongly suspect that the appveyor error is a docker problem with symlinks under Windows, related to:
[docker/compose#3902](https://github.com/docker/compose/issues/3902)

**Comment by xylar** [comment]
@chrisburr as in #7838, I'm inclined to ignore the problem on appveyor and hope for the best with the `noarch` build from `circleci`.  @lukaszlacinski, please let me know if you have any feedback on this.

File: recipes/globus-cli/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  license_file: LICENSE
+  summary: Globus CLI
+  description: A Command Line Interface to Globus.
+  doc_url: https://globus.github.io/globus-cli
+  dev_url: https://github.com/globusonline/globus-cli
+
+extra:
+  recipe-maintainers:
+    - xylar
+    - lukaszlacinski
```
Comment (chrisburr) [inline recipes/globus-cli/meta.yaml]: @lukaszlacinski Are you happy to be listed as a maintainer of this recipe?

Comment (lukaszlacinski) [reply recipes/globus-cli/meta.yaml]: Yes.

**Comment by xylar** [comment]
Thanks @chrisburr and @lukaszlacinski !

**Pull request merged**